### ISSUE TITLE
在GPRO训练中为什么进行两次前向传播分别取出 logits 和 aux_loss

### ISSUE THREAD

**Issue by Dxpsk** [jingyaogong/minimind#699]
##  问题：为什么在 GRPO 训练中要进行两次前向传播分别获取 `logits` 和 `aux_loss`？

您好，感谢这个项目的优秀实现！

GRPO 训练代码中，对于同一个输入 `outputs`，模型进行了**两次前向传播**：

```python
def get_per_token_logps(mdl, input_ids, n_keep):
    input_ids = input_ids.detach().clone() if input_ids.is_inference() else input_ids
    logits = mdl(input_ids, logits_to_keep=n_keep + 1).logits[:, :-1, :]
    per_token_logps = []
    for logits_row, ids_row in zip(logits, input_ids[:, -n_keep:]):
        ids_row = ids_row.detach().clone() if ids_row.is_inference() else ids_row
        per_token_logps.append(
            torch.gather(
                logits_row.log_softmax(dim=-1),
                1,
                ids_row.unsqueeze(1)
            ).squeeze(1)
        )
    return torch.stack(per_token_logps)

with autocast_ctx:
    per_token_logps = get_per_token_logps(model, outputs, completion_ids.size(1))  # [B*num_gen, R]
    res = model(outputs) if lm_config.use_moe else None
    aux_loss = res.aux_loss if res is not None else torch.tensor(0.0, device=args.device)
```

这里实际上对 `model(outputs)` 调用了两次：

- 在 `get_per_token_logps` 中，用于计算 logits → 用于策略梯度
- 再次调用 `model(outputs)`，用于获取 `aux_loss`（例如 MoE 的路由损失）

**Comment by jingyaogong** [comment]
感谢反馈，该问题先前已在新版本中修复，新版本近期更新
